pipeline_dp/combiners.py

```diff
@@ -225,56 +263,67 @@ def merge_accumulators(self, accumulator1: AccumulatorType,
                            accumulator2: AccumulatorType):
         return accumulator1 + accumulator2
 
-    def compute_metrics(self, accumulator: AccumulatorType) -> dict:
-        return {
-            'privacy_id_count':
-                dp_computations.compute_dp_count(
-                    accumulator, self._params.scalar_noise_params)
-        }
+    def compute_metrics(self, count: AccumulatorType) -> dict:
+        return {"privacy_id_count": self.get_mechanism().add_noise(count)}
```

Comment:
This seems to be a shift in architecturefrom one  where dp_computations were mostly responsible for dp logic to one where combiners actually do the dp logic but only use dp_computations to add noise.

Is supporting PLD inherently incompatible with using primitives for Sum/Count/etc. from the building block libraries?

Reply:
This is not shift from dp_computations, this refactoring for making dp_computations more OOP.
Namely it introduces nice OOP concept:
1. [AdditiveMechanism](https://github.com/OpenMined/PipelineDP/blob/28d30a020adf3e171c871ab7e86f0b0fcb08026a/pipeline_dp/dp_computations.py#L496), which encapsulate everythng related to dealing with mechanism.
2. [Sensitivities](https://github.com/OpenMined/PipelineDP/blob/28d30a020adf3e171c871ab7e86f0b0fcb08026a/pipeline_dp/dp_computations.py#L579) which keeps sensitivities and makes conversion from l0/linf -> l1/l2 sensitivities

And there are factory functions for creating `Sensitivities`/`AdditiveMechanism` for different situations. 

There are many benefits of having classes instead of plain functions, even w/o PLD, e.g. it simplifies testing of noise added by combiner, we can take mechanism from combiner and check that the mechanism is correct (noise_kind, noise_parameter), in the previous setup there were no easy method.

Supporting PLD will be simple - just needed to create another factory method for creating mechanism. 

Also this design plays nice with L1 contribution bounding - `Sensitivities` already supports l1 :).